app/src/free/java/org/mozilla/fenix/components/metrics/LeanplumMetricsServiceImpl.kt

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.metrics
+
+import android.app.Application
+import android.net.Uri
+import java.util.UUID
+
+class LeanplumMetricsServiceImpl(
```

Comment:
LeanplumMetricsService is used for two cases, both of which already have interfaces. Rather than making a dummy metrics service, let's configure those two locations.

First one is easy, the service is used in MetricController.create and discarded if the Telemetry flag is set to false. No change needed.

Second location is for the DeepLinkIntentProcessor. It wants a DeepLinkVerifier instance. You could pass in a dummy version that always returns false based on build flags.

Reply:
Sorry, I think I was getting confused with BuildConfig flags and build variants. I thought this could be accomplished with an if statement. 

Are we allowed to use any form of telemetry in the F-Droid version? If not, we may want to replace the entire MetricController. We have a dummy `DebugMetricController` already that could be used.

Reply:
> Are we allowed to use any form of telemetry in the F-Droid version?

I'm pretty sure that yes, telemetry is allowed as long as the frameworks being used are completely FLOSS (so Glean and Adjust should be OK, not firebase because of the dependency to gms). You can even display ads under the same conditions.

However, telemetry, especially when enabled by default like it is in Fenix, is considered an "anti-feature". This means there will be a warning prior downloading the app on the store. Such warning is already displayed for Fennec, so that's not too bad I guess.